src/sdk/product/useProductLink.ts

```diff
@@ -1,18 +1,54 @@
-import type { ProductSummary_ProductFragment } from '@generated/graphql'
-import type { CurrencyCode, ViewItemData, ViewItemEvent } from '@faststore/sdk'
-import { useSession, sendAnalyticsEvent } from '@faststore/sdk'
 import { useMemo } from 'react'
+import { useSession, sendAnalyticsEvent } from '@faststore/sdk'
+import type { ProductSummary_ProductFragment } from '@generated/graphql'
+import type {
+  CurrencyCode,
+  ViewItemData,
+  ViewItemEvent,
+  SelectItemEvent,
+  SelectItemData,
+  Item,
+} from '@faststore/sdk'
 
 export type ProductLinkOptions = {
   slug: string
+  index: number
 } & ProductSummary_ProductFragment
 
-export const useProductLink = ({ slug, ...product }: ProductLinkOptions) => {
+type CustomItem = Item & {
+  product_reference_id?: string
+}
+
+type GASelectItemEventData = SelectItemData & { items: CustomItem[] }
+
+interface GASelectItemEvent extends SelectItemEvent {
+  data: GASelectItemEventData
+}
+
+export const useProductLink = ({
+  slug,
+  index,
+  ...product
```

Comment:
I think it's better if you change the options here to accept only two properties: `index` and `product`. This way it's clearer what a "product" is and the type definitions are also easier to read. To get the slug instantly as a variable, you can do something like:

```suggestion
  product: { slug }
```

This doesn't change the product either remove the slug from it. It's just a shorthand to get the slug as a separate variable.